Feature update property

jListo opened the pull request:
Added the client side for update property feature. Will be changed to be only accessible by agents in the next update.

Commit:
Added Update Property Feature

Allows the owner and agent to update the image and tenant, this will
need to be limited to only the agent.

Commit:
Appearance Update

Updated the styles and text to be more appealing

nirrek commented on scripts/components/OwnerPropertyDetails.jsx:
```diff
@@ -38,6 +39,11 @@ var OwnerPropertyDetails = React.createClass({
           </div>
+          <div style={style.address}>
+            <UpdatePropertyForm 
+                propertyDetailsUpdated={this.getPropertyDetails} 
```
`this.getPropertyDetails` is not a defined method...

nirrek commented on scripts/components/UpdatePropertyForm.jsx:
```diff
@@ -0,0 +1,146 @@
+          image: '',
+        });
+
+        this.props.inspectionReportAdded();
```
`this.props.inspectionReportAdded()` is not a property that is passed to the component. This means that the submit callback is not occurring, and the form is also not being dismissed.

nirrek committed:
(OwnerPropertyDetails) fix usage of undefined function

nirrek committed:
(UpdatePropertyForm) fix undefined function invokation

nirrek commented:
This feature was non-functional btw. You were passing a non-existent function `getPropertyDetails` to the `<UpdatePropertyForm>` component. You were also invoking a non-existent function `inspectionReportAdded` on form submission. This meant that the form was not being dismissed, nor was any updated data being fetched.

A pull request should be a fully functional feature that is ready for deployment. This means you should have properly tested your feature. Please don't just submit a PR for half-finished stuff.

nirrek commented:
Stuff that needs doing:
- [ ] All details of the property need to be updateable.
- [ ] All existing data for the property needs to be prepopulated in the form, and should be submitted to the server.
- [ ] Clientside validation to ensure all fields have acceptable values prior to form submission.
- [ ] Cleanup state correctly when the dialog is closed. Eg. If there is an image size error, and you close/reopen dialog it is still sitting there.
- [ ] As with the add property form, there should be immediate feedback to the user if they enter an email for which a user does not exist.

Commit:
Merge branch 'master' into feature-update-property

Commit:
Update Property Expansion

Update the update property form to include more fields and fill fields
with current values.

Commit:
Validation for Update Property Form

Added validation and error messages for the update property form.

Commit:
Removed Photo String Size

Removed the photo string size from validation added due to
misconception.

Commit:
Add Photo

property photo will now display within the form.

Commit:
Revert "Add Photo"

This reverts commit 972893a4552814e5a5f8b2a70739aff93a96e11d.

Commit:
Reverted Accidental Revert

Accidentally reverted the previous commit, this reverts that.

Commit:
Added Server Errors and Minor Bug Fix

Added a message for server errors and fixed the missing photo reference.

Commit:
Error Display Bug Fix

Fixed the size error not disappearing on form submit/refresh and popup
close.

Commit:
Form Value Updates

Form values are now updated when the window pops up rather than when it
is loaded.

ghost commented:
Should be good now let me know about any issues, also it was already merged with master when I made my commits so it should be working form that front.

nirrek commented:
@jListo Again, same as the comment I left on your server PR, there are no new commits from you that address the issues. You probably have only made commits to your local repository.

nirrek committed:
(UpdatePropertyForm) allow tenantEmail to be null
